source/algorithms.tex

```diff
@@ -4219,6 +4249,81 @@
 \tcode{Searcher} need not meet the \oldconcept{CopyConstructible} requirements.
 \end{itemdescr}
 
+\rSec2[alg.starts.with]{Starts with}
+
+\indexlibraryglobal{starts_with}%
+\begin{itemdecl}
+template<@\libconcept{input_iterator}@ I1, @\libconcept{sentinel_for}@<I1> S1, @\libconcept{input_iterator}@ I2, @\libconcept{sentinel_for}@<I2> S2,
+         class Pred = ranges::equal_to, class Proj1 = identity, class Proj2 = identity>
+  requires @\libconcept{indirectly_comparable}@<I1, I2, Pred, Proj1, Proj2>
+  constexpr bool ranges::starts_with(I1 first1, S1 last1, I2 first2, S2 last2, Pred pred = {},
+                                     Proj1 proj1 = {}, Proj2 proj2 = {});
+template<@\libconcept{input_range}@ R1, @\libconcept{input_range}@ R2, class Pred = ranges::equal_to, class Proj1 = identity,
+         class Proj2 = identity>
+  requires @\libconcept{indirectly_comparable}@<iterator_t<R1>, iterator_t<R2>, Pred, Proj1, Proj2>
+  constexpr bool ranges::starts_with(R1&& r1, R2&& r2, Pred pred = {},
+                                     Proj1 proj1 = {}, Proj2 proj2 = {});
```

Comment:
@jwakely: I'm not sure why this second overload doesn't have a specification, but that's what's in the paper. Please let me know if anything is missing here (or if this is covered by some blanket wording elsewhere).

Reply:
The range overloads map to the first/last overloads per [algorithms.requirements] p14.